src/multisketch.rs

```diff
@@ -141,6 +152,92 @@ impl MultiSketch {
         log::trace!("s1_start:{s1_offset} s1_end:{}", s1_offset + s1_slice.len(),);
         s1_slice
     }
+
+    pub fn remove_sketches(&self, ids: &Vec<String>)
+    {   
+        // TODO: remove sketch bins which belong to the duplicate ids
+        todo!();
+    }
+
+    pub fn is_compatible_with(&self, sketch2: &Self) -> bool {
+        self.kmer_lengths() == sketch2.kmer_lengths()
+            && self.sketch_size == sketch2.sketch_size
+            && self.get_hash_type() == sketch2.get_hash_type()
+    }
+
+    pub fn merge_sketches(&self, sketch2: &Self) -> Self {
+        let mut sketch1_names: HashSet<String> = HashSet::new();
+        for sketch in self.sketch_metadata.iter() {
+            let name = sketch.name().to_string();
+            println!("{}", name);
+            sketch1_names.insert(name);
+        }
+
+        // First metadata
+        let mut merged_metadata = self.sketch_metadata.clone();
+        for sketch in sketch2.sketch_metadata.iter() {
+            if sketch1_names.contains(sketch.name()) {
+                let sample_name = sketch.name();
+                panic!("{sample_name} seems to appear in both databases. Cannot merge sketches.");
+            }
+            else {
+                merged_metadata.push(sketch.clone());
+            }
+        }
+        println!("Hello I am here");
+        // then merge sketches, create new multisketch instance
+        let mut merged_sketch = Self::new(&mut merged_metadata, self.sketch_size, &self.kmer_lengths, self.hash_type.clone());
+        println!("Empty merged sketch: {}", merged_sketch.sketch_bins.len());
+        // Merge actual sketche infos
+        merged_sketch.sketch_bins = self.sketch_bins.clone();
```

Comment:
Ideally this would just be streamed on disk, not all read into memory, then all written out. We can go through that together